warehouse/query-core/src/main/java/datawave/query/iterator/logic/OrIterator.java

```diff
@@ -15,54 +15,54 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 /**
  * Performs a deduping merge of iterators.
  *
- * NOTE***** if however sortedUIDs is false, then deduping is not performed and filters cannot be applied *****NOTE
  * 
  * @param <T>
  */
 public class OrIterator<T extends Comparable<T>> implements NestedIterator<T> {
     // temporary stores of uninitialized streams of iterators
-    private List<NestedIterator<T>> includes, excludes;
+    private List<NestedIterator<T>> includes, contextIncludes, contextExcludes;
     
     private Map<T,T> transforms;
     private Util.Transformer<T> transformer;
     
-    private TreeMultimap<T,NestedIterator<T>> includeHeads, excludeHeads;
-    
-    // sortedUIDs is normally true, however in some circumstances it may not in which case we cannot assume the underlying iterators are returning sorted
-    // UIDs. When this is true, we cannot advance iterators based on returned keys.
-    private final boolean sortedUIDs;
+    private TreeMultimap<T,NestedIterator<T>> includeHeads, contextIncludeHeads, contextIncludeNullHeads, contextExcludeHeads, contextExcludeNullHeads;
     
     private T prev;
     private T next;
     
     private Document prevDocument, document;
     
-    public OrIterator(Iterable<NestedIterator<T>> sources, boolean sortedUIDs) {
-        this(sources, null, sortedUIDs);
+    private T evaluationContext;
```

Comment:
that's fair

Reply:
I think the key here is that setContext can potentially be used even if context is not required. This can act as a hint, making it potentially applicable to all of the NestedIterators. At this time I am only implementing that for And/Or iterators but potentially could be used in all of them